packages/lesswrong/components/posts/AddToCalendar/AddToCalendarIcon.tsx

```diff
@@ -19,21 +20,41 @@ const styles = (theme: ThemeType): JssStyles => ({
 })
 
 const AddToCalendarIcon = ({post, label, hideTooltip, classes}: {
-  post: PostsList,
+  post: PostsList|PostsWithNavigation|PostsWithNavigationAndRevision,
   label?: string,
   hideTooltip?: boolean,
   classes: ClassesType,
 }) => {
-  // need both a start and end time to add an event to a calendar
-  if (!post.startTime || !post.endTime) {
+  // need both a start and end time to add an event to a calendar -
+  // if either are missing, we just return null
+  const missingStartOrEndTime = !post.startTime || !post.endTime;
+  
+  // we use the Facebook link as the default event details text
+  let eventDetails = post.facebookLink;
```

Comment:
[nit] I'd recommend moving this as close as possible to the other assignments so it's easily scan-able

Reply:
I actually did consider that, but I decided to keep it here so that I didn't have to rewrite my comments and explain why we are trying to get `plaintextDescription` first